src/api/app/policies/comment_policy.rb

```diff
@@ -31,4 +31,11 @@ def update?
   def reply?
     !(user.blank? || user.is_nobody? || record.user.is_nobody?)
   end
+
+  # Only logged-in Admins can moderate non-deleted comments
+  def moderate?
+    return false if user.blank? || user.is_nobody? || record.user.is_nobody?
+
+    user.is_admin?
```

Comment:
```suggestion
    return true if user.try(:is_admin?)
    
    false
```

Reply:
Alright, I applied your changes but added `&& !record.user.is_nobody?` to the condition to avoid moderating on comments that were soft-deleted already.